src/nexusrpc/handler/_core.py

```diff
@@ -401,3 +414,69 @@ def submit(
         self, fn: Callable[..., Any], *args: Any
     ) -> concurrent.futures.Future[Any]:
         return self._executor.submit(fn, *args)
+
+
+class OperationHandlerMiddleware(ABC):
+    """
+    Interceptor for operation handlers.
+
+    This should be extended by any operation handler interceptors.
+    """
+
+    def intercept(
```

Comment:
Since this is a single-method abstract class, I don't think there is value in use providing a default implementation. I think we should decorate this with `@abstractmethod` and make the body `...` which forces users to implement it.